script.js

```diff
@@ -1 +1,150 @@
+
+        console.log("valid");
+        console.log("Higher than 1900: " + (carYear.value > 1900));
+        console.log("Lower than current date: " + (carYear.value < currentDate.getFullYear()));
+        let statementValue = (parseInt(carYear.value) > 1900)  && (parseInt(carYear.value) < currentDate.getFullYear());
+        console.log("Value of if statement: " + statementValue);
+        console.log("Value: " + carYear.value);
+        console.log("Current Year: " + currentDate.getFullYear());
+        
+        carYear.parentElement.classList.add("input-valid");
+    
+    }else{
+        console.log("invalid");
+        console.log("Higher than 1900: " + (carYear.value > 1900));
+        console.log("Lower than current date: " + (carYear.value < currentDate.getFullYear()));
+        let statementValue = (parseInt(carYear.value) > 1900)  && (parseInt(carYear.value) < currentDate.getFullYear());
+        console.log("Value of if statement: " + statementValue);
+        console.log("Value: " + carYear.value);
+        console.log("Current Year: " + currentDate.getFullYear());
+
+        carYear.parentElement.classList.add("input-invalid");
+        carYear.setAttribute("title", "Not a valid car year");
+
+    }
+
+    if(parkingDayOfWeek === 0 || parkingDayOfWeek === 6){
+        console.log("Weekend: " + parkingDayOfWeek);
+
+        costPerDay = 7;
+        
+        console.log(costPerDay);
+
+    }else{
+        console.log("Weekday: " + parkingDayOfWeek);
+
+        costPerDay = 5;
+        
+        console.log(costPerDay)
+    }
+
+    if(!isNaN(Number(numDays.value)) &&
+       Number(numDays.value) < 30    &&
+       Number(numDays.value) > 1){
+
+        console.log("valid");
+
+        numDays.parentElement.classList.add("input-valid");
+        total.textContent = "The cost is $" + (Number(numDays.value))*costPerDay;
+    }else{
+        console.log("invalid");
+
+        carYear.parentElement.classList.add("input-invalid");
+
+    }
+
+    if(parkingDay > currentDate.getDay()     &&
+       parkingMonth > currentDate.getMonth() &&
+       parkingYear > currentDate.getFullYear()){
+        
+        time.parentElement.classList.add("input-valid");
+
+    }else{
+        time.parentElement.classList.add("input-invalid");
+        creditCard.setAttribute("title", "Not a date");
+
+    }
+
+    if(validateCardNumber(creditCard.value) === true){
+        console.log(validateCardNumber(creditCard.value));
+
+        creditCard.parentElement.classList.add("input-valid");
+
+    }else{
+        console.log(validateCardNumber(creditCard.value));
+
+        creditCard.parentElement.classList.add("input-invalid");
+        creditCard.setAttribute("title", "Not a valid card number");
+
+    }
+
+    if(expireMonth < currentDate.getMonth()   && 
+       expireYear < currentDate.getFullYear() &&
+       expireMonth < 12                       &&
+       (!isNaN(expireYear))                   &&
```

Comment:
If you did this check first, you could avoid doing the comparisons to the current date unnecessarily with bad data. This is a performance and efficiency point. First weed out the crappy data *then* do comparisons only on good data.